### ISSUE TITLE
gumbel-softmax temperature decay in wav2vec pretraining recipe 

### ISSUE THREAD

**Issue by thadeous** [speechbrain/speechbrain#1503]
It appears the Common Voice / self-supervised learning / wav2vec training recipe does not decay the gumbel-softmax temperature.  Instead, the value stays constant at 2 (incentivizing higher exploration in quantized code book sampling).  In a few experiments, it appears advantageous to follow a similar decay process as seen in the Huggingface Trainer: https://github.com/huggingface/transformers/blob/v4.20.1/examples/research_projects/wav2vec2/run_pretrain.py#L275-L282 

module.set_gumbel_temperature(
                max(self.max_gumbel_temp * self.gumbel_temp_decay**self.num_update_step, self.min_gumbel_temp), where defaults are:

max_gumbel_temp = 2
gumbel_temp_decay = .0.999995
min_gumbel_temp = 0.5

Should this decay be called here? https://github.com/speechbrain/speechbrain/blob/v0.5.12/recipes/CommonVoice/self-supervised-learning/wav2vec2/train.py#L125

**Comment by TParcollet** [comment]
HI! Very interesting, I was not aware of that decay. Yes, it should be called here. Maybe these parameters should be put in the yaml as w so they can be accessed with self.hparams. Any chance that you could do a PR for that?

**Comment by TParcollet** [comment]
Thanks @thadeous !!

**Comment by anautsch** [comment]
Closing this issue, as there is #1505 - please feel free to re-open when necessary :)
